src/safeposix/syscalls/net_calls.rs

```diff
@@ -3077,6 +3125,7 @@ impl Cage {
             );
         }
 
+        // get the flags
```

Comment:
can we add error checking here, such that we error out for invalid combination of flags, something like this: 

`       if socktype & !(SOCK_NONBLOCK | SOCK_CLOEXEC | 0x7) != 0 {
            return syscall_error(
                Errno::EOPNOTSUPP,
                "socket",
                "Invalid combination of flags"
            );
        }`